setup/gcp/installer/util/gcloud.py

```diff
@@ -827,13 +830,24 @@ def check_deployment_status(deployment_name, status):
         ['gcloud', 'deployment-manager', 'deployments', 'describe',
          deployment_name, '--format=json'])
 
+    deployment_err_msg = 'Error occurred during the deployment, exiting...'
+
     if return_code:
         print(err)
-        print('There is something wrong with the deployment, exiting...')
+        print(deployment_err_msg)
         sys.exit(1)
 
     deployment_info = json.loads(out)
 
-    current_status = deployment_info['deployment']['operation']['status']
+    deployment_operation = deployment_info['deployment']['operation']
+
+    deployment_status = deployment_operation['status']
+
+    deployment_error = deployment_operation.get('error', {})
+
+    if deployment_error:
+        print(deployment_error)
+        print(deployment_err_msg)
+        sys.exit(1)
```

Comment:
Is termination the right thing to do?  Or is it something recoverable that the user can try again, after the rest of the installation is finished.

Reply:
After some research on this topic, this is actually very hard to generalize since not all the errors are recoverable (for example, if you have an instance with the same name, the deployment will always fail no matter how many times you try it). I think by printing the error and exit the installation is a safe way of handling the exceptions for now since the user will have the knowledge of what happened and will be able to either file an issue with the error message or re install again if it's the error is not something serious. We should add more documentation around this area, let me update the error message to something more clear to the user knows when the db related issue occurred, they can try re running the installation process.